src/urbit/app/chess.hoon

```diff
@@ -765,7 +828,52 @@
               ::  add new games to our list
               %=  this
                 games  %+  ~(put by games)  u.game-id
-                       [game position |2.u.game-state]
+                       u.new.move-result
+              ==
+            %chess-result
+              =/  result
+                !<([game-id=@dau result=chess-result move=(unit chess-move)] q.cage.sign)
+              ?~  move.result
+                ?.  (check-threefold fen-repetition.u.game-state position.u.game-state)
+                  [~ this]  ::  nice try, cheater
+                :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update
+                          !>([%result u.game-id result.result])
+                      ==
+                      :*  %give  %kick  :~  /game/(scot %da u.game-id)/updates
+                                            /game/(scot %da u.game-id)/moves
+                                        ==
+                          ~
+                      ==
+                  ==
+                =/  updated-game  game.u.game-state
+                =.  result.updated-game  `result.result
+                %=  this
+                  games    (~(del by games) u.game-id)
+                  archive  (~(put by archive) u.game-id updated-game)
+                ==
+              =/  move-result
+                (try-move u.game-state (need move.result))
+              ?~  new.move-result
+                [cards.move-result this]  ::  nice try, cheater    
+              =,  u.new.move-result
+              ?.  (check-threefold fen-repetition position)
+                [~ this]  ::  nice try, cheater
+              :-  %+  weld  cards.move-result
+              ^-  (list card)
+              :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update
+                          !>([%result u.game-id result.result])
+                      ==
+                      :*  %give  %kick  :~  /game/(scot %da u.game-id)/updates
+                                            /game/(scot %da u.game-id)/moves
+                                        ==
+                          ~
+                      ==
+                  ==
+              %=  this
+                games    (~(del by games) u.game-id)
+                archive  (~(put by archive) u.game-id game)
```

Comment:
I believe there's a bug in the version you've written here that will archive a game without the last move when that move is sent with the `%chess-game-result`. The archive line should save the `game` from `result-game-state` rather than `u.game-state`.
```archive  (~(put by archive) u.game-id game(result `result.result))```

Reply:
Yes, you're correct - good catch!